### DISCUSSION TITLE
Using same images multiple times in convert (to GIF)

### DISCUSSION THREAD

**Discussion by loicaigon** [ImageMagick/ImageMagick#5938] [Development] [answered]
### ImageMagick version

6.9.3-7 Q16 x86_64 2016-03-27

### Operating system, version and so on

Mac OS 12.5.1

### Description

I am converting images to animated GIF file just fine with the convert tool. The problem I have is that I want in some case, reuse a file in the sequence of files to convert.

Let's say I have A.gif, B.gif and C.gif. I want my animation to be set like so:
A
B
C
C
B
A

So I end up with : 
convert -dispose 3 -delay 25 a.gif -delay 25 b.gif -delay 25 c.gif -delay 25 c.gif -delay 25 b.gif -delay 25 a.gif animation.gif

The problem is that the output animation is basically:
A
B
C
A
B
C

I tried to duplicate and rename each base file (ex: a.gif => a2.gif) and run the command:
convert -dispose 3 -delay 25 a.gif -delay 25 b.gif -delay 25 c.gif -delay 25 c2.gif -delay 25 b2.gif -delay 25 a2.gif animation.gif
A
B
C
A
B
C
Same result.

I tried to "touch" *2.gif files then run command but still…
A
B
C
A
B
C
…same result.

The only result I could get was to actually hard edit the files with Photoshop and run the command again.
Then the order is the one expected.
A
B
C
C
B
A

Disregarding my attempts, is there any reason why my first command is not producing the result I am expecting? And is there any correct approach to perform this?

Thanks in advance for your help,

Loic

**Comment by fmw42** [comment, marked as answer]
See -duplicate X where x is the number of duplicates of you want of the previous image.

A B C C B A    would be A B C -duplicate 1 B A

Is that what you need?

**Reply by loicaigon** [reply]
Thanks @fmw42 It looks interesting and I need to investigate. Basically sequence could have any arrangement "A B C C A A B C B C" and my first understanding is that duplicate send "copies" at the end of the list. So I need to do some testing and I will come back to you asap. But it seems a very interesting possible solution.

**Reply by fmw42** [reply]
You can use clones or mpr: 

\( A -write mpr:A \) \( B -write mpr:B \) \( C -write pr:C \) mpr:C mpr:A mpr:A mpr:B mpr:C mpr:B mpr:C

See
https://imagemagick.org/Usage/files/#mpr
https://imagemagick.org/Usage/basics/#clone

**Reply by loicaigon** [reply]
Thank you so much @fmw42 after a bit of a struggle I could achieve my goal thanks to MRPs and that's even solve another questioning of mine in the middle, so double score here :)

In case it helps others, the command that works was:
`convert -dispose 3 -loop 0 \( -delay 25 "a.gif" -write mpr:A \) \( -delay 25 "b.gif" -write mpr:B \) \( -delay 25 "c.gif" -write mpr:C \) -delay 25 mpr:C -delay 25 mpr:B -delay 25 mpr:A "output.gif"
`
For some reasons, escaping parenthesis was a prerequisite for the command to run fine.

**Comment by snibgo** [comment]
> So I end up with :
convert -dispose 3 -delay 25 a.gif -delay 25 b.gif -delay 25 c.gif -delay 25 c.gif -delay 25 b.gif -delay 25 a.gif animation.gif
The problem is that the output animation is basically:
A
B
C
A
B
C

If this is really occurring, it may be because you are using IM v6, which did not necessarily execute options in the order you gave them. I suggest you upgrade to IM v7.

**Reply by loicaigon** [reply]
Thanks @snibgo I will try to see if updating can help (for now updating fails on my machine). But with MRPs as @fmw42 suggested, I could get to achieve my goal.
